components/layout_2020/flow/inline.rs

```diff
@@ -174,29 +174,78 @@ impl<'box_tree, 'a, 'b> InlineFormattingContextState<'box_tree, 'a, 'b> {
         // >    property is normal, nowrap, or pre-line.
+        let mut whitespace_trimmed = Length::zero();
         for item in line_items.iter_mut().rev() {
-            if !item.trim_whitespace_at_end() {
+            if !item.trim_whitespace_at_end(&mut whitespace_trimmed) {
                 break;
             }
         }
 
+        let inline_start_position =
+            self.calculate_inline_start_for_current_line(containing_block, whitespace_trimmed);
         let mut state = LineItemLayoutState {
-            inline_position: self.next_line_start_position.inline,
+            inline_position: inline_start_position,
             max_block_size: Length::zero(),
-            containing_block_inline_start: self.next_line_start_position.inline,
+            inline_start_of_parent: Length::zero(),
             ifc_containing_block: containing_block,
             positioning_context: &mut self.positioning_context,
+            line_block_start: self.current_line_start_position.block,
         };
 
         let positioning_context_length = state.positioning_context.len();
-        let mut fragments = layout_line_items(line_items, layout_context, &mut state);
+        let fragments = layout_line_items(line_items, layout_context, &mut state);
 
+        let size = Vec2 {
+            inline: containing_block.inline_size,
+            block: state.max_block_size,
+        };
+
+        // The inline part of this start offset was taken into account when determining
+        // the inline start of the line in `calculate_inline_start_for_current_line` so
+        // we do not need to include it in the `start_corner` of the line's main Fragment.
+        let start_corner = Vec2 {
+            inline: Length::zero(),
+            block: self.current_line_start_position.block,
+        };
+        self.current_line_start_position = Vec2 {
+            inline: Length::zero(),
+            block: self.current_line_start_position.block + size.block,
+        };
+
+        if let Some(sequential_layout_state) = self.sequential_layout_state.as_mut() {
+            sequential_layout_state.advance_block_position(size.block);
+        }
+
+        let line_had_content =
+            !fragments.is_empty() || state.positioning_context.len() != positioning_context_length;
+        if line_had_content {
+            state
+                .positioning_context
+                .adjust_static_position_of_hoisted_fragments_with_offset(
+                    &start_corner,
+                    positioning_context_length,
+                );
+
+            self.fragments
+                .push(Fragment::Anonymous(AnonymousFragment::new(
+                    Rect { start_corner, size },
+                    fragments,
+                    containing_block.style.writing_mode,
+                )));
+        }
+    }
+
+    /// Given the amount of whitespace trimmed from the start of the line and taking into consideration
```

Comment:
The comment here says 'amount of whitespace trimmed from the *start of the line*' but in code, the value for whitespace_trimmed is set by the 'trim_whitespace_at_end` method. Is the comment here wrong or am I misunderstanding it?

Reply:
Oh, that's a good point. I've updated this to smiply say "the amount of whitespace trimmed from the line."